### PR TITLE
Feature: Improve Box Shadow Function + Improve Button Animation Rendering Performance  

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
http://vjira2:8080/browse/BDS-1663

## Summary
Updates the `bolt-shadow` function to allow other components to use the raised shadow styles (without forcing any extra styles to also get included). This update is then applied to the Button component to improve the rendering performance of the Button's hover / active animations.

## Details
- Adds a new optional param to the `bolt-shadow` function to allow the base set of shadow values to be changed to use the `raised` set of styles. Workaround till this entire function / mixin gets refactored.
- Cleans up the Button component's animation to use this updated function + revert back to using the similar performant animation technique I implemented in the original Button (https://tobiasahlin.com/blog/how-to-animate-box-shadow/)

## How to test
- [ ] Review code changes
- [ ] Test to make sure these animation-related CSS updates don't have any unforeseen side effects with the other button variations or have any cross browser issues when interacting with the button.

### CODE REVIEW

File: packages/components/bolt-button/src/_button-settings-and-tools.scss [hunk 1]
```diff
@@ -18,8 +18,8 @@ $bolt-button-border-radii: (
   'full': 50em,
 );
 $bolt-button-transition: $bolt-transition;
-$bolt-button-translate--hover: translate3d(0, -1px, 0);
-$bolt-button-translate--active: translate3d(0, 1px, 0);
+$bolt-button-translate--hover: translate3d(0, -2px, 0);
```
Comment (mikemai2awesome) [inline packages/components/bolt-button/src/_button-settings-and-tools.scss]: @sghoweri 

This translate effect is the possible cause of this button flashing bug (http://vjira2:8080/browse/BDS-1672) as pointed out by @danielamorse. The current `bolt-shadow` mixin is [also using translate similarly](https://github.com/bolt-design-system/bolt/blob/6f2a9eaedba0fc06cfd0a7db2bfb399afda51fd2/packages/core/styles/02-tools/tools-shadow/_tools-shadow.scss#L33).

We might need to rethink how we do the raised effects here.

Comment (mikemai2awesome) [reply packages/components/bolt-button/src/_button-settings-and-tools.scss]: Marking this resolved as the issue stated above was fixed by updating modal css.

File: packages/components/bolt-button/src/_button-settings-and-tools.scss [hunk 2]
```diff
@@ -68,23 +73,24 @@ $bolt-button-translate--active: translate3d(0, 1px, 0);
...
-    width: 100%;
-    height: 100%;
+    width: calc(100% + 2px); // since the border on the parent creates a 1px offset
+    height: calc(100% + 2px); // since the border on the parent creates a 1px offset
     pointer-events: none;
     border-radius: bolt-button-border-radius('regular');
-    @include bolt-shadow('level-10', true);
-    box-shadow: var(--bolt-button-box-shadow, #{bolt-shadow('level-10')});
-    will-change: box-shadow;
+    box-shadow: bolt-shadow('level-20', false, false, 'raised');
```
Comment (mikemai2awesome) [inline packages/components/bolt-button/src/_button-settings-and-tools.scss]: This is looking way more complicated than before. We should chat about this.

**Comment by mikemai2awesome** [comment]
I would like to fix the button without introducing new features to the `bolt-shadow` mixin and function, because they are not easy to use and the two are very inconsistent. They should both do the same thing but that's not the case right now.

So my preference is:

1. Quick fix to button
2. Refactor the shadow function and mixin later.

@sghoweri @danielamorse

**Comment by sghoweri** [comment]
> I would like to fix the button without introducing new features to the `bolt-shadow` mixin and function, because they are not easy to use and the two are very inconsistent. They should both do the same thing but that's not the case right now.
> 
> So my preference is:
> 
> 1. Quick fix to button
> 2. Refactor the shadow function and mixin later.
> 
> @sghoweri @danielamorse
 
+1 I'm good with this. Addresses the button updates we need for now without making more of a mess with the shadow function (easier to refactor laterr)

**Comment by mikemai2awesome** [comment]
Replaced by https://github.com/boltdesignsystem/bolt/pull/1900

**Pull request closed without merging**